dspace-api/src/main/java/org/dspace/storage/bitstore/S3BitStoreService.java

```diff
@@ -179,22 +180,32 @@ public void put(Bitstream bitstream, InputStream in) throws IOException {
         try {
             FileUtils.copyInputStreamToFile(in, scratchFile);
             long contentLength = scratchFile.length();
+            // The ETag may or may not be and MD5 digest of the object data.
+            // Therefore, we precalculate before uploading
+            String localChecksum = org.dspace.curate.Utils.checksum(scratchFile, "MD5");
```

Comment:
This should use the CSA constant instead of hardcoding "MD5" here. That will probably get rid of the lgtm warning as well.